source/extensions/filters/network/zookeeper_proxy/utils.h

```diff
@@ -27,20 +29,42 @@ class BufferHelper : public Logger::Loggable<Logger::Id::filter> {
 public:
   BufferHelper(uint32_t max_len) : max_len_(max_len) {}
 
-  int32_t peekInt32(Buffer::Instance& buffer, uint64_t& offset);
-  int64_t peekInt64(Buffer::Instance& buffer, uint64_t& offset);
-  std::string peekString(Buffer::Instance& buffer, uint64_t& offset);
-  bool peekBool(Buffer::Instance& buffer, uint64_t& offset);
+  absl::StatusOr<int32_t> peekInt32(Buffer::Instance& buffer, uint64_t& offset);
+  absl::StatusOr<int64_t> peekInt64(Buffer::Instance& buffer, uint64_t& offset);
+  absl::StatusOr<std::string> peekString(Buffer::Instance& buffer, uint64_t& offset);
+  absl::StatusOr<bool> peekBool(Buffer::Instance& buffer, uint64_t& offset);
   void skip(uint32_t len, uint64_t& offset);
   void reset() { current_ = 0; }
 
 private:
-  void ensureMaxLen(uint32_t size);
+  absl::Status ensureMaxLen(const uint32_t size);
+  absl::Status ensureMinLen(Buffer::Instance& buffer, uint64_t& offset, const uint32_t size);
```

Comment:
```suggestion
  absl::Status ensureMinLen(const Buffer::Instance& buffer, uint64_t offset, const uint32_t size);
```

I don't have preference on const offset or not so i'll defer to you

Reply:
Yeah, I will add `const` since `ensureMaxLen` does not need to change the `offset` value.